### DISCUSSION TITLE
Docker and unable to use a different database

### DISCUSSION THREAD

**Discussion by LucasMali** [evennia/evennia#2324] [Evennia Questions]
So I've been able to set up the docker instance using the image evennia/evennia with the default SQLlite, however, I'd like to move to Percona or MySQL equivalent. Every time I attempt to do so, I get the error that I don't have mysql-client installed **"Did you install mysqlclient?".** I believe that the container is based on Arch Linux and unable to install mysql-client via pacman. 

bash: pacman: command not found.

I went as far as trying to create an ubuntu instance (it works) but oddly enough I get the same error message mysql-client not found an error even though mysql-client is installed. So I'm thinking that I'm not placing the configurations in the proper location. 

game/server/secret_settings.py

based on this tutorial. https://github.com/evennia/evennia/wiki/Choosing-An-SQL-Server

Has anyone been able to get this to work and can guide me on the proper implementation using docker?

```
django.core.exceptions.ImproperlyConfigured: Error loading MySQLdb module.
```

**Comment by Griatch** [comment]
The docker image is based on python:3.7-alpine. Alpine uses the APK package manager, so something like this should be possible to use to get mysql client inside the docker container (you'd need to add it to your own custom copy if `Dockerfile`:

```
apk update
apk add mysql-client
```

You should first set up mysql locally on your machine (same way we set up sqlite locally), so it's not removed when the container dies. You also need to mount the mysql directory (using e.g. `VOLUME` in the `Dockerfile`) so it's available to the running processes inside the docker container.

**Reply by LucasMali** [reply]
So here is what I have.

_docker-compose_
```version: '2'
services:
  evennia:
    build: .
    container_name: evennia
    stdin_open: true
    tty: true
    # restart: always
    ports:
      - '4000:4000'
      - '4001:4001'
      - '4002:4002'
    environment:
      - GAME_NAME=${GAME_NAME}
    volumes:
      - './:/usr/src/game'
    networks:
      - default
  percona:
    image: percona
    container_name: db
    ports:
      - '3306:3306'
    environment:
      - MYSQL_ROOT_PASSWORD=root
    networks:
      - default
networks:
  default:
    driver: bridge
volumes:
  dbdata:
    driver: local
```

_Dockerfile_
```
FROM ubuntu:latest

ARG DEBIAN_FRONTEND=noninteractive

RUN apt-get -y -qq update; \
    apt-get -y upgrade; \
    apt-get install -y apt-utils; \
    apt-get install -y software-properties-common; \
    add-apt-repository ppa:deadsnakes/ppa -y

RUN apt-get -y -qq update; \
    apt-get install -y \
    python3.7 \
    python3-pip \
    python3.7-dev \
    python3-setuptools \
    virtualenv \
    gcc \
    git

WORKDIR /usr/src/game/

# FROM evennia/evennia:latest
# ENTRYPOINT evennia start -l
```
I'll try using `FROM evennia/evennia:latest` and running apk.

**Reply by LucasMali** [reply]
OK, I got this as a setup.

```
FROM evennia/evennia:latest
USER root
RUN apk update; \
    apk add mysql-client

WORKDIR /usr/src/game/
```

But even after setting it up, creating the database, validating that it can connect using `mysql -u root -proot -h db` and running the commands 
```
CREATE USER 'evennia'@'localhost' IDENTIFIED BY 'somepassword';
CREATE DATABASE evennia;
ALTER DATABASE `evennia` CHARACTER SET utf8;      # note that it's `evennia` not 'evennia'!
GRANT ALL PRIVILEGES ON evennia.* TO 'evennia'@'localhost';
FLUSH PRIVILEGES;
```
and this as my settings
```
DATABASES = {
   'default': {
       'ENGINE': 'django.db.backends.mysql',
       'NAME': 'evennia',
       'USER': 'evennia',
       'PASSWORD': 'somepassword',
       'HOST': 'db',  # or an IP Address that your DB is hosted on
       'PORT': '3306', # use default port
   }
}
```
I'm still stuck on this error after running `evennia migrate`
```
django.core.exceptions.ImproperlyConfigured: Error loading MySQLdb module.
Did you install mysqlclient?
```

**Reply by Griatch** [reply]
I think the issue may be that a python-lib for mysql is missing, not the normal client. Try adding `pip install mysqlclient` to your dockerfile and see if that helps.

**Reply by Griatch** [reply]
@LucasMali Did installing `mysqlclient` help?
